test/e2e/controller_test.go

```diff
@@ -0,0 +1,401 @@
+			t.Logf("not done waiting for CatalogEntry %q|%q to be reconciled", workspaceCluster, entryName)
+			return false, fmt.Errorf("CatalogEntry %q|%q hasn't been reconciled: %v", workspaceCluster, entryName, err)
+		}
+		return true, nil
+	}); err != nil {
+		return nil, fmt.Errorf("CatalogEntry %q|%q not found: %v", workspaceCluster, entryName, err)
+	}
+
+	return &catalogEntry, nil
+}
+
+const characters = "abcdefghijklmnopqrstuvwxyz"
+
+func randomName() string {
+	b := make([]byte, 10)
+	for i := range b {
+		b[i] = characters[rand.Intn(len(characters))]
+	}
+	return string(b)
+}
+
+// TestValidCatalogEntry verifies that our catalog controller can validate valid
+// catalogEntry by adding correct conditions, resources and exportPermissionClaims
+func TestValidCatalogEntry(t *testing.T) {
+	t.Run("testing valid catalog entry", func(t *testing.T) {
+		workspaceCluster := parentWorkspace(t).Join(randomName())
+		c := createWorkspace(t, workspaceCluster)
+
+		// Create test apiResourceSchema
+		schema := &apisv1alpha1.APIResourceSchema{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "today.tests.catalog.kcp.dev",
+			},
+			Spec: apisv1alpha1.APIResourceSchemaSpec{
+				Group: "catalog.kcp.dev",
+				Names: apiextensionsv1.CustomResourceDefinitionNames{
+					Plural:   "tests",
+					Singular: "test",
+					Kind:     "Test",
+					ListKind: "testlist",
+				},
+				Scope: apiextensionsv1.ClusterScoped,
+				Versions: []apisv1alpha1.APIResourceVersion{{
+					Name:    "v1",
+					Served:  true,
+					Storage: true,
+					Schema: runtime.RawExtension{
+						Raw: []byte(`{"description":"foo","type":"object"}`),
+					},
+				}},
+			},
+		}
+		err := createAPIResourceSchema(t, c, workspaceCluster, schema)
+		if err != nil {
+			t.Fatalf("failed to create APIResourceSchema in cluster %q: %v", workspaceCluster, err)
+		}
+
+		// Create test APIExport
+		export := &apisv1alpha1.APIExport{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "export-test",
+			},
+			Spec: apisv1alpha1.APIExportSpec{
+				LatestResourceSchemas: []string{"tests.catalog.kcp.dev"},
+				PermissionClaims: []apisv1alpha1.PermissionClaim{
+					{
+						GroupResource: apisv1alpha1.GroupResource{Resource: "configmaps"},
+					},
+				},
+			},
+		}
+		err = createAPIExport(t, c, workspaceCluster, export)
+		if err != nil {
+			t.Fatalf("failed to create APIExport in cluster %q: %v", workspaceCluster, err)
+		}
+
+		// Create catalogentry
+		newEntry := &catalogv1alpha1.CatalogEntry{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "entry-test",
+			},
+			Spec: catalogv1alpha1.CatalogEntrySpec{
+				Exports: []apisv1alpha1.ExportReference{
+					{
+						Workspace: &apisv1alpha1.WorkspaceExportReference{
+							Path:       workspaceCluster.String(),
+							ExportName: "export-test",
+						},
+					},
+				},
+			},
+		}
+		entry, err := createCatalogEntry(t, c, workspaceCluster, newEntry)
+		if err != nil {
+			t.Fatalf("unable to create CatalogEntry: %v", err)
+		}
+
```

Comment:
Insert polling loop to get entry and expect `conditions.IsTrue(entry, catalogv1alpha1.APIExportValidType)`